bitnami/fluentd/templates/aggregator-svc.yaml

```diff
@@ -1,32 +0,0 @@
-{{- if and .Values.aggregator.enabled .Values.aggregator.service.ports }}
```

Comment:
Why are we removing it?

Reply:
Because this Service is useless, the aggregators Pod are already selected and covered by the [headless service](https://github.com/irizzant/charts-1/blob/fluentd-istio/bitnami/fluentd/templates/svc-headless.yaml)

Reply:
Can you share more information on this? The closest I can find is this paragraph from [Istio's documentation](https://istio.io/docs/ops/deployment/requirements/):

> Service association: A pod must belong to at least one Kubernetes service even if the pod does NOT expose any port. If a pod belongs to multiple Kubernetes services, the services cannot use the same port number for different protocols, for instance HTTP and TCP.

But this shouldn't be an issue as both services are using the same ports and the same protocols.

The service was actually needed if you wanted to aggregate logs from the outside via TCP. This isn't possible anymore.

Reply:
> But this shouldn't be an issue as both services are using the same ports and the same protocols.

Yes but Envoy listeners are created on IP - port combinations basis and if there are multiple Service hitting the same IP - port combination Istio generates conflicts.

>The service was actually needed if you wanted to aggregate logs from the outside via TCP. This isn't possible anymore.

What do you mean by this? Having the aggregator Service would expose the aggregator port 24224 outside the cluster, which would allow to send logs to the aggregators Pod and that's fine.
Nevertheless, in a typical Kubernetes cluster having NodePort open is a problematic choice since the cluster nodes could not be directly reachable, which would require extra configuration at networking and loadbalancer levels.
If this use case is still needed for any reason, nothing prevents you from using the forwarder Service instead.
You just need to open the very same port 24224 and configure forwarders to send traffic to their aggregators.

Reply:
Hi @irizzant @dani8art , removal of this file disable the service creation of type "LoadBalancer" and "NodePort". Since this file does support service type "ClusterIP" (before the deletion), could you consider removing the headless service instead? It looks like the changes you that you done to the headless service is portable to this file. FYI, we are not using the headless service nor the port 24224 in my current project, but using NodePort service with another layer of load balancer setup. Thanks!

Reply:
Hi @ngyl88 
the headless Service is needed because aggregators are StatefulSet, they need persistence and they cannot be changed.
The forwarder Service does support ClusterIP, LoadBalancer and NodePort, so it supports also being deployed behind cloud providers, plus the forwarder configuration can be customized to forward logs to other services listening inside/outside the Kubernetes cluster.